File: 02 tests/ex3_sort.py
#!/usr/bin/env python3

""" ex3_sort.py

Sort 10^7 random integers

:author:	wolf
:created:	2025.05.22
"""


import time
import random

iterations = 3
timing_results = []

# define the test function
# --- function to be timed ---


def python_sort(n):
    arr = [random.randint(0, 100000) for _ in range(n)]
    arr.sort()
    return arr


for i in range(iterations):
    start = time.time()

    # call the test function
    python_sort(10**7)

    end = time.time()

    timing_results.append(end - start)


print(f"Elapsed time:  {sum(timing_results)/iterations:.4f} seconds")

File: 02 tests/ex3_sort.js
// run deno ex3_sort.js | node ex3_sort.js

/* ex3_sort.js

Sort 10^7 random integers

:author:	wolf
:created:	2025.05.22
*/


const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

// Array.from does not work with arrays size > 10^7
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length

function jsSort(n) {
    const arr = Array.from({ length: n }, () => Math.floor(Math.random() * 100000));
    return arr.sort((a, b) => a - b);
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsSort(10**7);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Elapsed time: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000, "seconds");